cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -81,3 +80,132 @@ func (s *Server) GetPackageRepositories(ctx context.Context, request *corev1.Get
+						if reasonString, ok := conditionAsMap["reason"]; ok && reasonString == "IndexationSucceed" {
+							ready = true
+							break
+						}
+					}
+				}
+			}
+		}
+
+		if !ready {
+			log.Infof("Skipping packages for repository [%s] because it has not reached 'Ready' state:n%v", name, repoUnstructured.Object)
+		} else {
+			url, found, err := unstructured.NestedString(repoUnstructured.Object, "status", "url")
+			if err != nil || !found {
+				log.Infof("field status.url not found on HelmRepository: %w:\n%v", err, repoUnstructured.Object)
+				continue
+			}
+
+			log.Infof("Found repository: [%s], index url: [%s]", name, url)
+			repoPackages, err := readPackagesFromRepoIndex(url)
+			if err != nil {
+				// just skip this repo
+				log.Errorf("Failed to read packages for repository [%s] due to %v", name, err)
+			} else {
+				responsePackages = append(responsePackages, repoPackages...)
+			}
+		}
+	}
+	return &corev1.GetAvailablePackagesResponse{
+		Packages: responsePackages,
+	}, nil
+}
+
+func getHelmRepos(ctx context.Context) (*unstructured.UnstructuredList, error) {
+	// TODO: replace incluster config with the user config using token from request meta.
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("unable to create incluster config: %w", err)
+	}
+
+	client, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create dynamic client: %w", err)
+	}
+
+	repositoryResource := schema.GroupVersionResource{
+		Group:    fluxGroup,
+		Version:  fluxVersion,
+		Resource: fluxHelmRepoResource}
+
+	// Currently checks globally. Update to handle namespaced requests (?)
+	repos, err := client.Resource(repositoryResource).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("unable to list fluxv2 helmrepositories: %w", err)
+	} else {
+		// TODO: should we filter out those repos that don't have .status.condition.Ready == True?
+		// like we do in GetAvailablePackages()?
+		// i.e. should GetAvailableRepos() call semantics be such that only "Ready" repos are returned
+		return repos, nil
+	}
+}
+
+func readPackagesFromRepoIndex(indexURL string) ([]*corev1.AvailablePackage, error) {
+	//Get the response bytes from the url
+	response, err := http.Get(indexURL)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != 200 {
+		return nil, fmt.Errorf("received non 200 response code: [%d]", response.StatusCode)
+	}
+
+	contents, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var index helmrepo.IndexFile
+	err = yaml.Unmarshal(contents, &index)
+	if err != nil {
+		return nil, err
+	}
+
+	index.SortEntries()
+	responsePackages := []*corev1.AvailablePackage{}
+	for _, entry := range index.Entries {
+		if entry[0].GetDeprecated() {
+			log.Infof("skipping deprecated chart: [%s]", entry[0].Name)
+			continue
+		}
+		pkg := &corev1.AvailablePackage{
+			Name:    entry[0].Name,
```

Comment:
`entry[0]` - assuming that's the latest version? Nice. We can always query for more versions when looking at the package detail, I assume.

Reply:
yeah, this is by no means the final iteration what this will look like, just a first take based on looking existing code